microprofile/WFLY-12590_Integrate_MicroProfile_Fault_Tolerance_2.0.adoc

```diff
@@ -0,0 +1,102 @@
+
+* https://thorntail.io/[Thorntail] has integrated MicroProfile Fault Tolerance as a https://github.com/thorntail/thorntail/tree/master/fractions/microprofile/microprofile-fault-tolerance[fraction].
+* https://quarkus.io/[Quarkus] has integrated MicroProfile Fault Tolerance as an https://github.com/quarkusio/quarkus/tree/master/extensions/smallrye-fault-tolerance[extension].
+
+== Requirements
+
+* Support Eclipse MicroProfile Fault Tolerance version 2.0 for all applications deployed in WildFly.
+* [WILL BE CLARIFIED] The extension is either added to all configurations provided by WildFly (standalone and domain mode)
+  or provided as a layer in a separate build profile.
+* Upstream Eclipse MicroProfile Fault Tolerance TCK test suite will be added to `microprofile-tck` WildFly upstream test suite.
+
+== Nice-to-Have Requirements
+
+* [WILL BE CLARIFIED] Create a MicroProfile Fault Tolerance https://github.com/wildfly/quickstart[quickstart].
+
+Items that are currently only being taken into consideration with next version of this feature:
+
+* The underlying library used by SmallRye FT implementation is Hystrix which is currently in "maintenance" mode and a final
+  release of the project (version 1.5.18) has been released. The discussion about replacing the library is ongoing.
+* WildFly's standard management model will be configuration-less
+** all configuration is done through MicroProfile Config specification
+** given the maintenance problem above it is not desirable to expose a management model which will change in near future as concepts are not cleanly mappable to a different underlying library
+* Support for plurality of isolated deployments.
+
+== Non-Requirements
+
+* N/A
+
+== Implementation Plan
+
+* Add dependencies on `smallrye-fault-tolerance` and `microprofile-fault-tolerance-api` artifacts.
+* Add a new `microprofile-fault-tolerance-smallrye` extension to provide Fault Tolerance APIs support for deployments.
+* Add configuration-less MicroProfile Fault Tolerance subsystem.
+* Support configuration of Hystrix from MicroProfile Config specifications. The configuration is queried for Hystrix properties which are then applied through Archaius configuration mechanism.
+
+== Test Plan
+
+* The `smallrye-fault-tolerance` component is passing the MicroProfile Fault Tolerance TCK during its release process.
+* WildFly integration test suite will be enhanced with tests that deploys Java archives which uses the MicroProfile Fault Tolerance API.
+
+== Community Documentation
+
+The feature will be documented in WildFly Administration Guide in a new MicroProfile Fault Tolerance section.
+Since WildFly integration does not impose any special requirements and subsystem configuration, this section will just contain how to configure Hystrix and link to MP FT specification.
+This section will also describe migration path from Thorntail primarily regarding Hystrix configuration.
```

Comment:
Will be TT properties supported too ?

https://docs.thorntail.io/2.5.0.Final/#_hystrix

Reply:
That's not a plan – we have never supported configuration of different application servers/runtimes in WildFly.